extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/Sheets.pulldown/Rename PDF Sheets.pushbutton/script.py

```diff
@@ -1,44 +1,59 @@
-import os
-import sys
+# -*- coding: utf-8 -*-
 import os.path as op
+from pathlib import Path
 
-from pyrevit import UI
 from pyrevit import forms
 
+def renamepdf(old_name):
+    new_name = prefix_stripper.sub("", old_name)
+    new_name = capitalizer.sub(lambda m : "-{}".format(m.group(1).upper()), new_name)
+    new_name = normalizer.sub("-", new_name)
+    if not new_name:
+        raise ValueError("Renaming results in an empty string.")
+    return new_name
 
 # if user shift-clicks, default to user desktop,
 # otherwise ask for a folder containing the PDF files
 if __shiftclick__:
-    basefolder = op.expandvars('%userprofile%\\desktop')
+    basefolder = op.expandvars(r"%userprofile%\desktop")
 else:
     basefolder = forms.pick_folder()
-
-
-def renamePDF(pdffile):
-    import re
-    r = re.compile('(?<=Sheet - )(.+)')
-    fname = r.findall(pdffile)[0]
-    r = re.compile('(.+)\s-\s(.+)')
-    fnameList = r.findall(fname)
-    return fnameList[0][0] + ' - ' + fnameList[0][1].upper()
-
-
-if basefolder:
-    sheetcount = 0
-
-    # list files and find the PDF files in the base folder
-    filenames = os.listdir(basefolder)
-    for pdffile in filenames:
-        ext = op.splitext(pdffile)[1].upper()
-        if ext == '.PDF' and ('Sheet' in pdffile):
-            # if PDF make a new file name and rename the exisitng PDF file
-            newfile = renamePDF(pdffile)
-            try:
-                os.rename(op.join(basefolder, pdffile),
-                          op.join(basefolder, newfile))
-                sheetcount += 1
-            except Exception as e:
-                print("Unexpected error:", sys.exc_info()[0])
-
-    # let user know how many sheets have been renames
-    forms.alert('{0} FILES RENAMED.'.format(sheetcount))
+# exist script if no folder selected
+if not basefolder:
+    forms.alert("No Folder Selected.", exitscript=True)
+
+# Continue Script if basefolder
+import re
+prefix_stripper = re.compile(r"^.*?Sheet\s*-\s*")   #"Sheet -", with space management
+capitalizer = re.compile(r"-(?!.*-)\s*(.*)")        #Capitalized after the last hyphen
+normalizer = re.compile("\s*-\s*")                  #Normalize all other hyphens surrounded by spaces
+pdf_count, sheet_count, rename_count, err_count = 0, 0, 0, 0
+
+# Ask for sub folder treatment and difine a research patern
+dir_pattern = "**/" if forms.alert("Do you want to include subfolders?", yes=True, no=True) else ""
+
+#Search all .pdf file
+for pdf_file in Path(basefolder).glob("{}*.pdf".format(dir_pattern)):
+    pdf_count += 1
+    # do nothing if not a 'Sheet - ' pdf (test with the regex prefixe stripper use for renaminf file)
```

Comment:
comments should not be used to explain the implementation, but are fine to explain something not obvious:
```suggestion
    # do nothing if not a 'Sheet - ' pdf
```